File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) 
    {
        int count0{0}, count1{0}, count2{0};
        int n = nums.size();

        for(int i = 0; i < n; i++)
        {
            if(nums[i] == 0) count0++;
            else if(nums[i] == 1) count1++;
            else if(nums[i] == 2) count2++;
            else return;
        }
        int i = 0;
        while(count0--) nums[i++] = 0;
        while(count1--) nums[i++] = 1;
        while(count2--) nums[i++] = 2;
    }
};

File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int cnt0 = 0, cnt1 = 0, cnt2 = 0;

        for(int num : nums) {
            if(num == 0) cnt0++;
            else if(num == 1) cnt1++;
            else cnt2++;
        }

        int idx = 0;
        while(cnt0-->0) nums[idx++] = 0;
        while(cnt1-->0) nums[idx++] = 1;
        while(cnt2-->0) nums[idx++] = 2;
    }
}